Source/NSOrderedSet.m

```diff
@@ -0,0 +1,1903 @@
+   Created: May 17 2019
+
+   This file is part of the GNUstep Base Library.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02111 USA.
+
+*/
+#import "common.h"
+#import "Foundation/NSArray.h"
+#import "Foundation/NSAutoreleasePool.h"
+#import "Foundation/NSOrderedSet.h"
+#import "Foundation/NSCoder.h"
+#import "Foundation/NSArray.h"
+#import "Foundation/NSEnumerator.h"
+#import "Foundation/NSKeyValueCoding.h"
+#import "Foundation/NSValue.h"
+#import "Foundation/NSException.h"
+#import "Foundation/NSPredicate.h"
+#import "Foundation/NSLock.h"
+#import "Foundation/NSKeyValueObserving.h"
+
+#import <GNUstepBase/GSBlocks.h>
+#import "Foundation/NSKeyedArchiver.h"
+#import "GSPrivate.h"
+#import "GSFastEnumeration.h"
+#import "GSDispatch.h"
+#import "GSSorting.h"
+
+@class	GSOrderedSet;
+@interface GSOrderedSet : NSObject	// Help the compiler
+@end
+
+@class	GSMutableOrderedSet;
+@interface GSMutableOrderedSet : NSObject	// Help the compiler
+- (void) _raiseRangeExceptionWithIndex: (NSUInteger)index from: (SEL)sel;
+@end
+
+@implementation NSOrderedSet
+
+static Class NSOrderedSet_abstract_class;
+static Class NSMutableOrderedSet_abstract_class;
+static Class NSOrderedSet_concrete_class;
+static Class NSMutableOrderedSet_concrete_class;
+
+static SEL	addSel;
+static SEL	countSel;
+static SEL	eqSel;
+static SEL	oaiSel;
+static SEL	remSel;
+static SEL	rlSel;
+
++ (id) allocWithZone: (NSZone*)z
+{
+  if (self == NSOrderedSet_abstract_class)
+    {
+      return NSAllocateObject(NSOrderedSet_concrete_class, 0, z);
+    }
+  else
+    {
+      return NSAllocateObject(self, 0, z);
+    }
+}
+
++ (void) initialize
+{
+  if (self == [NSOrderedSet class])
+    {
+      [self setVersion: 1];
+      
+      addSel = @selector(addObject:);
+      countSel = @selector(count);
+      eqSel = @selector(isEqual:);
+      oaiSel = @selector(objectAtIndex:);
+      remSel = @selector(removeObjectAtIndex:);
+      rlSel = @selector(removeLastObject);
+
+      NSOrderedSet_abstract_class = self;
+      NSOrderedSet_concrete_class = [GSOrderedSet class];
+      [NSMutableSet class];
+    }
+}
+
+- (Class) classForCoder
+{
+  return NSOrderedSet_abstract_class;
+}
+
+// NSCoding
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  /*
```

Comment:
Could we remove this unused code?

Reply:
Done.

Reply:
Sorry, but the commented out code is still there.

Reply:
GOne now... I'm assuming you removed it.